### PR TITLE
vo: sanitize vsync values

### PR DESCRIPTION
**Pull request by kasper93**

### CODE REVIEW

File: video/out/vo.c [hunk 1]
```diff
@@ -513,6 +513,14 @@ static void update_vsync_timing_after_swap(struct vo *vo,
     if (in->num_successive_vsyncs <= 2)
         return;
 
+    if (vsync_time <= 0 || vsync_time <= prev_vsync) {
```
Comment (Dudemanguy) [inline video/out/vo.c:516]: I mean this is isn't harmful or anything, but is this needed? I just went with the approach of "don't give vo_sync_info nonsensical values" which seems reasonable enough.

Comment (kasper93) [reply video/out/vo.c:516]: I see 
```
        // Make up some crap if presentation feedback is missing.
        if (vsync.last_queue_display_time < 0)
            vsync.last_queue_display_time = mp_time_ns();
```
Should this be `<=`? Well I will see, but there is disconnect between d3d11 code and this logic currently.

Comment (Dudemanguy) [reply video/out/vo.c:516]: Technically yes, but mp_time_us() can never return 0 (if it does we're in some real trouble).

Comment (kasper93) [reply video/out/vo.c:516]: What do you think now? I still sanitize, but define 0 as invalid value. I think it is good thing to do?

Comment (Dudemanguy) [reply video/out/vo.c:516]: Kind of nitpicky here but I don't believe there's really a meaningful difference between 0 or -1? Why not just make 0 and below the new "invalid"?

Comment (kasper93) [reply video/out/vo.c:516]: ok

Comment (kasper93) [reply video/out/vo.c:516]: updated.

**Review by Dudemanguy — APPROVED** [review]

**Pull request merged**